spec/spec.md

```diff
@@ -0,0 +1,601 @@
+
+1. Cryptographic primitives and key management
+2. [[ref:Decentralized Identifiers]] (DIDs) and DID methods
+3. [[ref:Verifiable Credentials]] and Presentations and accompanying functionality
+4. Protocols for making use of [[ref:Decentralized Identifiers]] and [[ref:Verifiable Credentials]]
+
+::: note
+At present, [Decentralized Web Nodes](https://identity.foundation/decentralized-web-node/spec/), and other forms of decentralized storage, are out of scope of this specification.
+:::
+
+## Conformance
+
+As well as sections marked as non-normative, all authoring guidelines, diagrams, examples, and notes in this specification are non-normative. Everything else in this specification is normative.
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in [[spec:RFC2119]].
+
+## Terminology
+
+This specification uses the following terms:
+
+[[def:Decentralized Identifier, Decentralized Identifiers, DID, DIDs, DID Document, DID Documents]]
+~ A [W3C specification](https://www.w3.org/TR/did-core/) describing an identifier that enables verifiable, decentralized digital identity. A DID identifier is associated with a DID Document containing keys, services, and other data. 
+
+[[def:Verifiable Credential, Verifiable Credentials, VC, VCs]]
+~ A cryptographically secure, privacy-respecting, and machine-verifiable digital credential. There are many such data models in the W3C, ISO, IETF, and other standards development organizations (SDOs).
+
+[[def:Verifiable Presentation, Verifiable Presentations, VP, VPs]]
+~ A cryptographically secure, privacy-respecting, and machine-verifiable digital credential that contains an additional signature from a _holder_ who is authorized to present the credential to a relying party.
+
+[[def:Web5 SDK, Web5 SDKs]]
+~ A software development kit that provides developers with the tools and APIs necessary to build Web5-compatible applications.
+
+[[def:DID Resolver, DID Resolvers, DID Resolution]]
+~ A DID Resolver is a software service designed to perform resolution of a [[ref:DID Document]], which is the process of obtaining a [[ref:DID Document]] for a given [[ref:DID]]. See [[ref:DID-IDENTIFIER-RESOLUTION]].
+
+## Web5 Core Features
+
+The Web5 SDK provides a comprehensive set of features for building decentralized protocols and applications. The following section outline the core features required to build a conformant [[ref:Web5 SDK]].
+
+### Cryptographic Primitives
+
+Web5 supports the following cryptographic key types for use with the noted corresponding digital signature algorithms:
+
+| Key Type | Algorithm | Function |
+|----------|-----------|----------|
+| [[ref:secp256k1]] | `ES256K` [[spec:RFC8812]] | Signing and Verification |
+| [[ref:Ed25519]] | `EdDSA` [[spec:RFC8032]] | Signing and Verification |
```

Comment:
FWIW I was planning on adding support for the four key types from here https://did-dht.com/registry/index.html#key-type-index

Reply:
excellent, I'll add these